src/pages/CompanyListPage/index.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import Button from '@/components/Button';
+import CompanyItem from '@/components/CompanyItem';
+import TextInput from '@/components/TextInput';
+import { COLORS } from '@/constants/colors';
+import { FONT } from '@/constants/font';
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
+import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp';
+import {
+  Box,
+  Divider,
+  IconButton,
+  Pagination,
+  Typography,
+} from '@mui/material';
+
+interface Company {
+  id: number;
+  name: string;
+  email: string;
+}
+
+const CompanyListPage: React.FC = () => {
+  const { t } = useTranslation();
+  const [searchTerm, setSearchTerm] = useState<string>('');
+
+  const [sortOrder, setSortOrder] = useState<'asc' | 'desc'>('asc');
+
+  const companies: Company[] = [
+    { id: 1, name: 'Company Alpha', email: 'alpha@company.com' },
+    { id: 2, name: 'Company Beta', email: 'beta@company.com' },
+    { id: 3, name: 'Company Gamma', email: 'gamma@company.com' },
+  ];
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+    setSearchTerm(value);
+  };
+
+  const sortedAndFilteredCompanies = useMemo(() => {
+    const filtered = companies.filter((company) =>
+      company.name.toLowerCase().includes(searchTerm.toLowerCase())
+    );
+
+    const sorted = filtered.sort((a, b) => {
+      const nameA = a.name.toLowerCase();
+      const nameB = b.name.toLowerCase();
+
+      if (nameA < nameB) return sortOrder === 'asc' ? -1 : 1;
+      if (nameA > nameB) return sortOrder === 'asc' ? 1 : -1;
+      return 0;
+    });
+
+    return sorted;
+  }, [companies, searchTerm, sortOrder]);
```

Comment:
this should be done on be, as there is pagination, you just need to make right request

Reply:
will do.

Reply:
it shill there

Reply:
sorry, my bad. i'll be more attentive.